crates/apollo-compiler/test_data/diagnostics/0074_merge_identical_fields.txt

```diff
@@ -8,15 +8,15 @@ Error: operation must not select different types using the same field name `name
     │   ──┬─  
     │     ╰─── but the same field name has type `String!` here
 ────╯
-Error: operation must not select different types using the same field name `name`
+Error: operation must not select different fields to the same alias `name`
     ╭─[0074_merge_identical_fields.graphql:19:3]
     │
  18 │   name: nickname
     │   ───────┬──────  
-    │          ╰──────── `name` has type `String` here
+    │          ╰──────── field `name` is selected from field `nickname` here
  19 │   name
     │   ──┬─  
-    │     ╰─── but the same field name has type `String!` here
+    │     ╰─── but the same field `name` is also selected from field `name` here
```

Comment:
I've tried a slightly different approach here, also to address the "but" issue. Now explaining why it's invalid in the help message and unconditionally referring to the output name / response name / response key as the "alias" (you can only get into this situation with at least one alias as far as I understand, since we already know that the parent types overlap)
```
Error: cannot select multiple fields into the same alias `fido`
   ╭─[query.graphql:3:3]
   │
 2 │   fido: name
   │   ─────┬────  
   │        ╰────── `fido` is selected from `Dog.name` here
 3 │   fido: nickname
   │   ───────┬──────  
   │          ╰──────── `fido` is selected from `Dog.nickname` here
   │ 
   │ Help: Both fields may be present on the schema type, so it's not clear which one should be used to fill the response
───╯
```